src/main/starlark/builtins_bzl/common/cc/cc_helper.bzl

```diff
@@ -564,23 +565,36 @@ def _collect_native_cc_libraries(deps, libraries):
     transitive_libraries = [dep[CcInfo].transitive_native_libraries() for dep in deps if CcInfo in dep]
     return CcNativeLibraryInfo(libraries_to_link = depset(direct = libraries, transitive = transitive_libraries))
 
-def _tool_path(cc_toolchain, tool):
-    return cc_toolchain._tool_paths.get(tool, None)
+_TOOL_NAMES_TO_ACTION_NAMES = {
+    "ar": ACTION_NAMES.cpp_link_static_library,
+    "gcc": ACTION_NAMES.c_compile,
+    "ld": ACTION_NAMES.cpp_link_executable,
+    "objcopy": ACTION_NAMES.objcopy_embed_data,
+    "strip": ACTION_NAMES.strip,
+}
+
+def _tool_or_action_path(cc_toolchain, feature_configuration, tool):
```

Comment:
Could we just migrate this entirely to get_tool_for_action? Which things ultimately will still rely on _tool_paths? Could we add action_configs for those?

Reply:
Existing toolchains that do not specify action_configs rely on the tool paths. The default toolchains in `cc_rules` other than for MSVC on Windows do not specify action_configs. 

I would also like to see tool paths fully deprecated and removed in favour of action_configs, but that seems like a significant deprecation that would need proper messaging, and a bunch of prep work to ensure no disruptions for existing toolchains. This change unblocks toolchains that use action_configs getting the proper make variables before all that work is complete.